src/sentry/monitors/logic/mark_failed.py

```diff
@@ -304,6 +326,42 @@ def create_issue_platform_occurrence(
     )
 
 
+def get_failure_reason(failed_checkins: Sequence[SimpleCheckIn]):
+    """
+    Builds a humam readible strong from a list of failed check-ins.
+
+    "3 missed check-ins detected"
+    "2 missed check-ins, 1 timeout check-in and 1 error check-in were detected"
+    "A failed check-in was detected"
+    """
+    human_names: Mapping[int, str] = dict(
+        [
+            (CheckInStatus.ERROR, "error"),
+            (CheckInStatus.MISSED, "missed"),
+            (CheckInStatus.TIMEOUT, "timeout"),
+        ]
+    )
+
+    status_counts = Counter(
+        human_names[cast(int, checkin["status"])]
+        for checkin in failed_checkins
+        if checkin["status"] in human_names.keys()
+    )
+
+    if sum(status_counts.values()) == 1:
+        return _("A %(status)s check-in was detected") % {"status": list(status_counts.keys())[0]}
```

Comment:
Can we say "**An** error check-in was detected"?

Reply:
👍

Reply:
Done